frontend/src/components/App/TopBar.tsx

```diff
@@ -139,9 +139,21 @@ function AppBarActionsMenu({
       appBarActions.map(action => {
         const Action = has(action, 'action') ? action.action : action;
         if (React.isValidElement(Action)) {
+          const label = (action as AppBarAction)?.label;
           return (
             <ErrorBoundary>
-              <MenuItem>{Action}</MenuItem>
+              <MenuItem
+                onClick={(action as AppBarAction)?.onClick || undefined}
+                sx={{
+                  justifyContent: 'center',
+                  gap: '12px',
+                }}
+              >
+                <ListItemIcon sx={{ minWidth: '40px', justifyContent: 'center' }}>
+                  {Action}
+                </ListItemIcon>
```

Comment:
I am a bit puzzled. What are rendering here?

Reply:
With this code:

![image](https://github.com/user-attachments/assets/c762f93b-c76b-4ba5-a8a0-69833a628a75)

With the code commented out:

![image](https://github.com/user-attachments/assets/97ef3257-0295-486f-99fb-8fe37e5f74b0)